### PR TITLE
Adapt to SECURITY-170

### PR DESCRIPTION
**Pull request by rsandell**
By using an EnvironmentContributor instead of build parameters

@reviewbybees

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/registry/notification/webhook/PushNotification.java [hunk 1]
```diff
@@ -73,7 +75,25 @@ public int hashCode() {
...
 
-    abstract public Set<ParameterValue> getJobParamerers();
+    /**
+     * Provide parameters to be put into a build.
+     * @return the parameters
+     * @deprecated misspelled and wrong context naming. Use {@link #getRunParameters()}
+     */
+    @Deprecated
+    public Set<ParameterValue> getJobParamerers() {
+        if (Util.isOverridden(PushNotification.class, getClass(), "getRunParameters")) {
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/registry/notification/webhook/PushNotification.java]: 🐜 Minore code duplication

File: src/main/java/org/jenkinsci/plugins/registry/notification/EnvContributor.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import java.io.IOException;
+import java.util.Set;
+
+/**
+ * Provides environment variables to builds triggered by the plugin.
+ *
+ * @author Robert Sandell &lt;rsandell@cloudbees.com&gt;.
+ */
+@Extension
+public class EnvContributor extends EnvironmentContributor {
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/registry/notification/EnvContributor.java]: I would prefer EnvironmentContributingAction (global performance reason)

Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/registry/notification/EnvContributor.java]: Please add <code>@Restricted</code> just to support switch in the future

**Comment by oleg-nenashev** [comment]
🐜 till the EnvironmentContributor becomes Restricted, 🐝 after it

File: src/main/java/org/jenkinsci/plugins/registry/notification/webhook/PushNotification.java (hunk 1 above)
Comment (rsandell) [inline src/main/java/org/jenkinsci/plugins/registry/notification/webhook/PushNotification.java]: Where?

File: src/main/java/org/jenkinsci/plugins/registry/notification/EnvContributor.java (hunk 2 above)
Comment (rsandell) [inline src/main/java/org/jenkinsci/plugins/registry/notification/EnvContributor.java]: `EnvironmentContributingAction` is only applicable for `AbstractBuild`s

**Comment by oleg-nenashev** [comment]
@rsandell 👍

**Pull request merged**